docs/configuration.md

```diff
@@ -1,47 +1,160 @@
+calls faster.
+
+The `pushback_max_dedupe_in_flight` flag rate limits how many concurrent `add-*`
+requests identified as duplicates can be processed by the
+`3.` **synchronous** process that fetches entries and extracts information required
+to build SCTs. When this value is exceeded, TesseraCT returns `429 - Too Many Requests`
+to subsequent **duplicate** `add-*` requests only.  Non-duplicate `add-*`
+requests are not impacted, and can still be processed.  This limits the number of
+resources TesseraCT can spend on duplicate requests.
+
+## Chain filtering
+
+Chains accepted by TesseraCT can be filtered by setting the following flags:
+
+- `root_pem_file`: Path to the file containing root certificates that are
+acceptable to the log. Chains MUST be signed by one of the roots included in
+this file.
+- `reject_expired`: If true, TesseraCT rejects expired certificates.
+- `reject_unexpired`: If true, TesseraCT rejects certificates that are either
+currently valid or not yet valid.
+- `ext_key_usage`: A list of comma separated [Extended Key Usages (EKU) from x509](https://pkg.go.dev/crypto/x509#ExtKeyUsage).
+Certificates MUST include one or more of these EKUs to be accepted. If empty, no
+filtering is applied.
+- `reject_extension`: A comma separated list of X.509 extension OIDs, in dotted
+string form (e.g. '2.3.4.5'). Certificates that include one or more of these extensions
+will be rejected.
+- `not_after_start`: Start of the range of acceptable NotAfter values,
+inclusive. Leaving this unset or empty implies no lower bound to the range.
+RFC3339 UTC format, e.g: 2024-01-02T15:04:05Z.
+- `not_after_limit`: Cut off point of notAfter dates - only notAfter dates
+strictly *before* notAfterLimit will be accepted. Leaving this unset or empty
+means no upper bound on the accepted range. RFC3339 UTC format, e.g:
+2024-01-02T15:04:05Z.
+
+## Origin and submission prefix
+
+The origin used in Checkpoints is specified with the `origin` flag.
+The submission prefix of a log has two parts: the host and the serving path:
+`https://$HOST/$PATH_PREFIX/ct/v1/...`.
+
+As per [static-ct-api specs](https://c2sp.org/static-ct-api), `the origin line
+MUST be the submission prefix of the log as a schema-less URL with no trailing
+slashes`. Use your frontend serving infrastructure to make sure that requests
+to these URLs are correctly routed to a TesseraCT server. TesseraCT will serve
+requests it receives regardless of their `$HOST`. However, it will match the
+$PATH_PREFIX, with the `path_prefix` flag.
+
+## Memory considerations
+
+TesseraCT's memory footprint is directly impacted by:
+
+- `inmemory_antispam_cache_size`: the number of entries from the antispam
+database cached locally
+- `batch_max_size`: the number of entries that are kept in memory before
+sequenced in a batch
+- [The number of cached issuers keys](https://github.com/transparency-dev/tesseract/blob/main/storage/storage.go)
+
```

Comment:
It's also heavily affected by the number of concurrent requests (because of things like context, parsed X.509 chains, goroutine stacks and overhead etc.), so `enable_awaiter` will be a "large lever" here.